### PR TITLE
Catch NotFoundException before `woocommerce_get_batch_processor`

### LINKED ISSUE #46974: woocommerce_get_batch_processor prevented from running when NotFoundException is thrown
### Prerequisites

- [x] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

[The inline documentation](https://github.com/woocommerce/woocommerce/blob/3d8e20c62cf1811610178d675212b6a23675661f/plugins/woocommerce/src/Internal/BatchProcessing/BatchProcessingController.php#L10) for setting up a custom batch procesor, says you can pass a processor instance to the `woocommerce_get_batch_processor`.

However, that filter will never run for a 3rd party processor [as the line above it ](https://github.com/woocommerce/woocommerce/blob/3d8e20c62cf1811610178d675212b6a23675661f/plugins/woocommerce/src/Internal/BatchProcessing/BatchProcessingController.php#L277) will look for the processor in the Container and not finding it, throw a `NotFoundException` exception.

If you replace the 

```
$processor = wc_get_container()->get( $processor_class_name );
```

```
try {
	$processor = wc_get_container()->get( $processor_class_name );
} catch ( \Exception $e) {
	$processor = null;
}
```

Then the filter is allowed to run.

### Expected behavior

Schedule action is completed and my processor runs

### Actual behavior

Scheduled action fails. 

### Steps to reproduce

I have tried to create an [example plugin](https://gist.github.com/helgatheviking/605bce4ae6197e0363d73b92b54e36d2) to help test. It's a very stripped down version of what I am working on so it might have some typos.

In theory it will add a tool in WooCommerce > Status > Tools. Running the tool should fail as the described bug should throw an error that the processor can't be found in the woo container. The action will be logged as failed by the action scheduler.

When my proposed change is enabled the action should complete and the Log should show `process a batch!`

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by helgatheviking**
### Changes proposed in this Pull Request:

Catchthing the NotFoundException allows the `woocommerce_get_batch_processor` filter to run for 3rd party plugins looking to use batch processing.

Closes #46974.

### How to test the changes in this Pull Request:
See #46974 for instructions

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

**Comment by senadir** [comment]
Also tagged Nestor who worked on that code.

**Comment by helgatheviking** [comment]
This is blocking the [anonymize woo tool](https://github.com/helgatheviking/anonymize-woocommerce) I'm working on. If my example doesn't work that is atleast partially working, but obviously comes with a big warning about being very destructive.

File: plugins/woocommerce/src/Internal/BatchProcessing/BatchProcessingController.php [hunk 1]
```diff
@@ -274,7 +274,12 @@ public function is_scheduled( string $processor_class_name ): bool {
 	 * @throws \Exception If it's not possible to get an instance of the class.
 	 */
 	private function get_processor_instance( string $processor_class_name ) : BatchProcessorInterface {
-		$processor = wc_get_container()->get( $processor_class_name );
+
+		try {
+			$processor = wc_get_container()->get( $processor_class_name );
+		} catch ( \Exception $e) {
+			$processor = null;
+		}
```
Comment (Konamiman) [inline plugins/woocommerce/src/Internal/BatchProcessing/BatchProcessingController.php]: We can avoid the exception altogether (and still get an useful `null`) if we use the container's `has` method instead:

```suggestion
		$container = $wc_get_container();
		$processor = $container->has( $processor_class_name ) ? $container->get( $processor_class_name ) : null;
```

**Review by Konamiman — CHANGES_REQUESTED** [review]
Hi, thanks for your contribution, I've left a change suggestion.

File: plugins/woocommerce/src/Internal/BatchProcessing/BatchProcessingController.php (hunk 1 above)
Comment (naman03malhotra) [reply plugins/woocommerce/src/Internal/BatchProcessing/BatchProcessingController.php]: Thanks for working on it, I agree with @Konamiman's suggestion here, if we can prevent exception all together by using `$wc_get_container()->has` method, that would be great

Comment (helgatheviking) [reply plugins/woocommerce/src/Internal/BatchProcessing/BatchProcessingController.php]: Just to confirm you do mean `wc_get_container()` and _not_ `$wc_get_container()` right? Commit incoming...

Comment (Konamiman) [reply plugins/woocommerce/src/Internal/BatchProcessing/BatchProcessingController.php]: Indeed, silly typo, sorry

**Review by Konamiman — APPROVED** [review]

**Review by naman03malhotra — APPROVED** [review]
I've tested the change, and found that the action is scheduled. Thanks for the test plugin, it was helpful while testing.

**Comment by helgatheviking** [comment]
@naman03malhotra is this targeted for 9.0 then?

**Comment by Konamiman** [comment]
@helgatheviking Yes, this change will be part of WooCommerce 9.0.

**Pull request merged**